gc/base/MemorySubSpaceGenerational.cpp

```diff
@@ -323,9 +323,18 @@ MM_MemorySubSpaceGenerational::counterBalanceContract(
 }
 
 uintptr_t
-MM_MemorySubSpaceGenerational::releaseFreeMemoryPages(MM_EnvironmentBase* env)
+MM_MemorySubSpaceGenerational::releaseFreeMemoryPages(MM_EnvironmentBase* env, uintptr_t memoryType)
 {
-	return _memorySubSpaceOld->releaseFreeMemoryPages(env);
+	uintptr_t releasedPages = 0;
+
+	Assert_MM_true_internal(OMR_ARE_ALL_BITS_SET(memoryType, MEMORY_TYPE_OLD));
```

Comment:
there is simpler Assert_MM_true (not even sure what is the point of _internal variant)

you could join the line what defines releasedPages  and when old pages are added

for clarity we could call it releasedPagesInBytes (someone may think it's page count)